### PR TITLE
ProxyThread SocketTimeoutException Verbosity

### PR DESCRIPTION
**Pull request by kingthorin**
Only log full exception if debug is enabled.

Fixes zaproxy/zaproxy#3095

### CODE REVIEW

File: src/org/parosproxy/paros/core/proxy/ProxyThread.java [hunk 1]
```diff
@@ -270,7 +271,10 @@ public void run() {
 	    	if (firstHeader != null) {
 	    		log.warn("Timeout accessing " + firstHeader.getURI());
 	    	} else {
-	    		log.warn("Timeout", e);
+	    		log.warn("Socket Timeout");
```
Comment (thc202) [inline src/org/parosproxy/paros/core/proxy/ProxyThread.java]: Worth changing to "Timed out while reading first message.", this is a special case.

File: src/org/parosproxy/paros/core/proxy/ProxyThread.java [hunk 2]
```diff
@@ -270,7 +271,10 @@ public void run() {
 	    	if (firstHeader != null) {
 	    		log.warn("Timeout accessing " + firstHeader.getURI());
 	    	} else {
-	    		log.warn("Timeout", e);
+	    		log.warn("Socket Timeout");
+	    		if (log.isDebugEnabled()) {
+	    			log.debug(e);
```
Comment (thc202) [inline src/org/parosproxy/paros/core/proxy/ProxyThread.java]: This should call `debug(e, e)` to include the exception's stack trace.

**Comment by kingthorin** [comment]
Done & done

**Comment by thc202** [comment]
Thank you! :)

**Pull request merged**